repos/system_upgrade/common/libraries/dnfplugin.py

```diff
@@ -214,10 +215,25 @@ def _transaction(context, stage, target_repoids, tasks, plugin_info, test=False,
             )
         except CalledProcessError as e:
             api.current_logger().error('DNF execution failed: ')
-            raise StopActorExecutionError(
-                message='DNF execution failed with non zero exit code.\nSTDOUT:\n{stdout}\nSTDERR:\n{stderr}'.format(
-                    stdout=e.stdout, stderr=e.stderr)
-            )
+
+            message = 'DNF execution failed with non zero exit code.'
+            details = {'STDOUT': e.stdout, 'STDERR': e.stderr}
+
+            if 'more space needed on the' in e.stderr:
+                # The stderr contains this error summary:
+                # Disk Requirements:
+                #   At least <size> more space needed on the <path> filesystem.
+
+                article_section = 'Generic case'
+                if xfs_info.present and xfs_info.without_ftype:
+                    article_section = 'XFS ftype=0 case'
+
+                message = ('There is not enough space on the file system hosting /var/lib/leapp directory '
+                           'to extract the packages.')
+                details = {'hint': "Please follow the instructions in the '{}' section of the article at: "
+                                   "link: https://access.redhat.com/solutions/5057391".format(article_section)}
+
+            raise StopActorExecutionError(message=message, details=details)
```

Comment:
It would be great to cover this with a unit test as well

Reply:
On the other hand - looking at the fact that we have zero unit tests for the _transaction and there is a great deal of mocking to do, I think we could live with manual testing + an integration test for this part. WDYT?

Reply:
Yes, as discussed on IRC, I agree with integration and manual testing due to the increased complexity of unit testing in this case.